org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/StringFragment.java

```diff
@@ -0,0 +1,181 @@
+	 * @since 3.0
+	 */
+	public static final SimplePropertyDescriptor ESCAPED_VALUE_PROPERTY =
+		new SimplePropertyDescriptor(StringFragment.class, "escapedValue", String.class, MANDATORY); //$NON-NLS-1$
+
+	/**
+	 * A list of property descriptors (element type:
+	 * {@link StructuralPropertyDescriptor}),
+	 * or null if uninitialized.
+	 */
+	private static final List PROPERTY_DESCRIPTORS;
+
+	static {
+		List propertyList = new ArrayList(2);
+		createPropertyList(StringFragment.class, propertyList);
+		addProperty(ESCAPED_VALUE_PROPERTY, propertyList);
+		PROPERTY_DESCRIPTORS = reapPropertyList(propertyList);
+	}
+
+	/**
+	 * Returns a list of structural property descriptors for this node type.
+	 * Clients must not modify the result.
+	 *
+	 * @param apiLevel the API level; one of the
+	 * <code>AST.JLS*</code> constants
+
+	 * @return a list of property descriptors (element type:
+	 * {@link StructuralPropertyDescriptor})
+	 * @since 3.0
+	 */
+	public static List propertyDescriptors(int apiLevel) {
+		return PROPERTY_DESCRIPTORS;
+	}
+
+	/**
+	 * The literal string; defaults to the
+	 * literal for the empty string.
+	 */
+	private String escapedValue = "";//$NON-NLS-1$
+
+
+
+	/**
+	 * Creates a new unparented string literal node owned by the given AST.
+	 * By default, the string literal denotes the empty string.
+	 * <p>
+	 * N.B. This constructor is package-private.
+	 * </p>
+	 *
+	 * @param ast the AST that is to own this node
+	 */
+	StringFragment(AST ast) {
+		super(ast);
+	}
+
+	@Override
+	final List internalStructuralPropertiesForType(int apiLevel) {
+		return propertyDescriptors(apiLevel);
+	}
+
+	@Override
+	final Object internalGetSetObjectProperty(SimplePropertyDescriptor property, boolean get, Object value) {
+		if (property == ESCAPED_VALUE_PROPERTY) {
+			if (get) {
+				return getEscapedValue();
+			} else {
+				setEscapedValue((String) value);
+				return null;
+			}
+		}
+		// allow default implementation to flag the error
+		return super.internalGetSetObjectProperty(property, get, value);
+	}
+
+	@Override
+	final int getNodeType0() {
+		return STRING_FRAGMENT;
+	}
+
+	@Override
+	ASTNode clone0(AST target) {
+		StringFragment result = new StringFragment(target);
+		result.setSourceRange(getStartPosition(), getLength());
+		result.setEscapedValue(getEscapedValue());
+		return result;
+	}
+
+	@Override
+	final boolean subtreeMatch0(ASTMatcher matcher, Object other) {
+		// dispatch to correct overloaded match method
+		return matcher.match(this, other);
+	}
+
+	@Override
+	void accept0(ASTVisitor visitor) {
+		visitor.visit(this);
+		visitor.endVisit(this);
+	}
+
+	/**
+	 * Returns the string value of this literal node to the given string
+	 * fragment token. The token is the sequence of characters that would appear
+	 * in the source program, excluding enclosing double quotes.
+	 *
+	 * @return the string fragment token, excluding enclosing double quotes
+	 */
```

Comment:
Quick check - in case of standard 21 without preview, will the StringFragment be returned ? [am yet to check the ASTCoverter - disclaimer]. If not, then we will need to have @noreference here since this is a preview feature